web/Controllers/LandlordController.cs

```diff
@@ -85,13 +82,13 @@ public async Task<ActionResult> Profile([FromRoute] int id)
             return StatusCode(403);
         }
 
-        var landlord = await _dbContext.Landlords.SingleOrDefaultAsync(l => l.Id == id);
+        var landlord = _landlordService.GetLandlordFromId(id);
         if (landlord == null)

```

Comment:
The `landlord` variable here is a `Task`, it will never be null. You should await it first.

```suggestion
        var landlord = await _landlordService.GetLandlordFromId(id);
        if (landlord == null)
```